RFS/src/main/java/com/rfs/transformers/Transformer_ES_6_8_to_OS_2_11.java

```diff
@@ -47,21 +59,38 @@ public ObjectNode transformGlobalMetadata(ObjectNode root) {
         componentTemplatesRoot.set("component_template", componentTemplatesSubRoot);
         newRoot.set("component_template", componentTemplatesRoot);
 
-        return newRoot;
+        return new GlobalMetadataData_OS_2_11(newRoot);
     }
 
     @Override
-    public ObjectNode transformIndexMetadata(ObjectNode root){
-        ObjectNode newRoot = root.deepCopy();
+    public IndexMetadata transformIndexMetadata(IndexMetadata index) {
+        var copy = index.deepCopy();
+        transformIndex(copy, IndexType.Concrete);
+        return copy;
+    }
 
-        TransformFunctions.removeIntermediateMappingsLevels(newRoot);
+    private void transformIndex(Index index, IndexType type) {
+        logger.debug("Original Object: " + index.rawJson().toString());
+        var newRoot = index.rawJson();
+
+        switch (type) {
+            case Concrete:
+                indexTransformations.forEach(transformer -> transformer.applyTransformation(index));
+                break;
+            case Template:
+                indexTemplateTransformations.forEach(transformer -> transformer.applyTransformation(index));
+                break;
+        }
 
         newRoot.set("settings", TransformFunctions.convertFlatSettingsToTree((ObjectNode) newRoot.get("settings")));
         TransformFunctions.removeIntermediateIndexSettingsLevel(newRoot); // run before fixNumberOfReplicas
         TransformFunctions.fixReplicasForDimensionality(newRoot, awarenessAttributeDimensionality);
 
-        logger.debug("Original Object: " + root.toString());
         logger.debug("Transformed Object: " + newRoot.toString());
-        return newRoot;
-    }    
+    }
+
+    private enum IndexType {
+        Concrete,
+        Template;
+    }
```

Comment:
Not going to block on this, but - is this the correct home for this enumerated type?  I'm guessing you'll be moving this to a more centralized location as soon as you need to do a different transformer.

Reply:
Yea - this is a good candidate to refactor over time